### PR TITLE
Add a raw bitmap lookup function

### PR DESCRIPTION
**Pull request by harendra-kumar**
Got some more performance boost by:
* Removing dynamic word size handling. We now use a static 64-bit word size. We
  can possibly add a 32-bit version as well and statically select an
  appropriate version based on WORD_SIZE_IN_BITS GHC macro.
* Removing testBit which caused an additional branch instruction
* Refactor `isCombining` such that GHC can reuse code converting char to int

@Bodigrim please take a look.

### CODE REVIEW

File: Data/Unicode/Internal/Bits.hs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        uncheckedIShiftRL#, uncheckedShiftL#)
+
+-- | @lookup64 addr index@ looks up the bit stored at bit index @index@ using a
+-- bitmap starting at the address @addr@. Looks up the 64-bit word containing
+-- the bit and then the bit in that word. The caller must make sure that the
+-- 64-bit word at the byte address (addr + index / 64) * 8 is legally
+-- accessible memory.
+--
+lookupBit64 :: Addr# -> Int -> Bool
+lookupBit64 addr# (I# index#) = W# (word## `and#` bitMask##) /= 0
```
Comment (Bodigrim) [inline Data/Unicode/Internal/Bits.hs:30]: Maybe `isTrue# (word2Int# (word## `and#` bitMask##))`? Just to be absolutely sure that there are no allocations.

File: Data/Unicode/Properties/CombiningClass.hs [hunk 2]
```diff
@@ -877,12 +876,14 @@ getCombiningClass _ = 0
 
 {-# INLINE isCombining #-}
 isCombining :: Char -> Bool
-isCombining c | (ord c) < 768 || (ord c) > 125258 = False
-isCombining c = combiningBitmap (ord c)
+isCombining c =
+    let i = ord c
+    in if i < 768 || i > 125258
+       then False
+       else combiningBitmap i
```
Comment (Bodigrim) [inline Data/Unicode/Properties/CombiningClass.hs:883]: Is there a reason to prefer this to `i >= 768 && i <= 125258 && combiningBitmap i`?

Comment (harendra-kumar) [reply Data/Unicode/Properties/CombiningClass.hs:883]: I don't prefer anyone in particular.

**Comment by Bodigrim** [comment]
Looks good to me :)

**Comment by harendra-kumar** [comment]
Thanks! Merging this.

**Pull request merged**